File: event-service/src/main/java/org/tggc/eventservice/repository/EventRepository.java
package org.tggc.eventservice.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;
import org.tggc.eventservice.model.Event;

import java.time.LocalDateTime;
import java.util.List;

@Repository
public interface EventRepository extends JpaRepository<Event, Long>, JpaSpecificationExecutor<Event> {

    List<Event> findByCreatorId(Long userId);

    List<Event> findByEventDateBetween(LocalDateTime now, LocalDateTime soon);
}

File: event-service/src/main/kotlin/org/tggc/eventservice/repository/EventRepository.kt
package org.tggc.eventservice.repository

import org.springframework.data.r2dbc.repository.R2dbcRepository
import org.springframework.stereotype.Repository
import org.tggc.eventservice.model.Event
import reactor.core.publisher.Flux
import java.time.LocalDateTime

@Repository
interface EventRepository : R2dbcRepository<Event, Long> {

    fun findByCreatorId(userId: Long): Flux<Event>

    fun findByEventDateBetween(now: LocalDateTime, soon: LocalDateTime): Flux<Event>
}
